gnomad_qc/v4/annotations/generate_freq.py

```diff
@@ -101,106 +119,84 @@ def get_vds_for_freq(
 
     logger.info("Annotating VDS with only necessary sample metadata...")
     vds = hl.vds.VariantDataset(
         vds.reference_data,
-        vds.variant_data.annotate_cols(
-            pop=meta_ht[vds.variant_data.col_key].population_inference.pop,
-            sex_karyotype=meta_ht[
-                vds.variant_data.col_key
-            ].sex_imputation.sex_karyotype,
-            fixed_homalt_model=meta_ht[
-                vds.variant_data.col_key
-            ].project_meta.fixed_homalt_model,
-            gatk_version=meta_ht[vds.variant_data.col_key].project_meta.gatk_version,
-            age=meta_ht[vds.variant_data.col_key].project_meta.age,
-            sample_age_bin=get_sample_age_bin(
-                meta_ht[vds.variant_data.col_key].project_meta.age
-            ),
+        vds.variant_data.select_cols(
+            pop=vds.variant_data.meta.population_inference.pop,
+            sex_karyotype=vds.variant_data.meta.sex_imputation.sex_karyotype,
+            fixed_homalt_model=vds.variant_data.meta.project_meta.fixed_homalt_model,
+            gatk_version=vds.variant_data.meta.project_meta.gatk_version,
+            age=vds.variant_data.meta.project_meta.age,
+            sample_age_bin=get_sample_age_bin(vds.variant_data.meta.project_meta.age),
             ukb_sample=hl.if_else(
-                meta_ht[vds.variant_data.col_key].project_meta.ukb_sample,
-                "ukb",
-                "non_ukb",
+                vds.variant_data.meta.project_meta.ukb_sample, "ukb", "non_ukb"
             ),
         ),
     )
 
+    # TODO: What above function? Also I think this can be done at a different time.
     # Downsamplings are done outside the above function as we need to annotate
-    # globals and rows
+    # globals and rows.
     logger.info("Annotating downsampling groups...")
-    vds = hl.vds.VariantDataset(
-        vds.reference_data,
-        annotate_downsamplings(vds.variant_data, pop_expr=vds.variant_data.pop),
+    vds.variant_data = annotate_downsamplings(
+        vds.variant_data, DOWNSAMPLINGS["v4"], pop_expr=vds.variant_data.pop
     )
 
+    # TODO: This is only used once and is not a complicated expression, do you think it
+    #  needs to be a function?
     logger.info("Annotating non_ref hets pre-split...")
-    vds = annotate_non_ref_het(vds)
+    vds.variant_data = vds.variant_data.annotate_entries(
+        _het_non_ref=vds.variant_data.LGT.is_het_non_ref()
+    )
 
     logger.info("Selecting only required fields to reduce memory usage...")
-    vds = hl.vds.VariantDataset(
-        vds.reference_data,
-        vds.variant_data.select_entries("LA", "LAD", "DP", "GQ", "LGT", "_het_non_ref"),
+    vds.variant_data = vds.variant_data.select_entries(
+        "LA", "LAD", "DP", "GQ", "LGT", "_het_non_ref"
     )
 
-    return vds
-
-
-def annotate_non_ref_het(vds: hl.vds.VariantDataset) -> hl.vds.VariantDataset:
```

Comment:
This is only used once and is not a complicated expression, do you think it needs to be a function?